### PR TITLE
Add 7.0 migration guide

### PR DESCRIPTION
**Pull request by JackLivio**
Adds notes about changes made to SDL Core that will impact the HMI integration. Noting I would like to include links to relevant documentation after the release is made and we have permanent links.

### CODE REVIEW

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    </description>
+    <param name="appID" type="Integer" mandatory="true">
+        <description>ID of application that is related to this RPC.</description>
+    </param>
+</function>
+```
+
+### Webengine Projection Support
+
+In SDL Core 6.1 Webengine apps was added as a way to connect an SDL Connected app to the head unit using a headless web browser on the head unit. This feature was expanded in SDL Core 7.0 With the addition of a new HMI_TYPE and template layout `WEB_VIEW`. Apps that use this HMI_TYPE and layout can present their own non-templated layout using HTML and javascript.
```
Comment (iCollin) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:220]: WEB_VIEW only applies to web engine, but it's not too related other than that. I would maybe remove some of the prelude. Here are some things I think could be added:

A new `WEB_VIEW` appHMIType and template layout was added which will allow apps to render a template independent view in a browser environment with javascript and HTML. 
This appHMIType must be explicitly specified in an app's policy table entry for SDL Core to allow it to be used.

**Review by iCollin — APPROVED** [review]

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
+# Updating from 6.1 to 7.0
+
+The 7.0 release had a number of changes and additions to the HMI API that will require updates to your SDL Core integration in your head unit. 
+
+## Environment Update
+
+The minimum environment requirements have changed for Ubuntu 18. GCC Version 7.5.x is now recommended over the previously recommended GCC Version 7.3.x
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:7]:
```suggestion
The minimum environment requirements have changed for Ubuntu 18. GCC Version 7.5.x is now recommended over the previously recommended GCC Version 7.3.x.
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+- The `url` parameter in the HMI API and RPC Spec has removed its max length requirement.
+
+- The parameter `appID` has been removed from VehicleInfo.UnsubscribeVehicleData request to the HMI.
+
+## Newly Deprecated
+
+### Deprecated Character Sets
+
+Character sets `TYPE2SET`, `TYPE5SET`, `CID1SET`, and `CID2SET` have been deprecated. These character sets only had proprietary significance and HMI's can now choose from the following character sets:
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:19]:
```suggestion
Character sets `TYPE2SET`, `TYPE5SET`, `CID1SET`, and `CID2SET` have been deprecated. These character sets only had proprietary significance and HMIs can now choose from the following character sets:
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+### HMI UI Additions
+
+#### Menu Changes
+
+SDL Core 7.0 adds extended capabilities to the app menu. SDL Core now supports nested submenus, dynamic menus, and menu browsing limitations while driver distraction mode is enabled. 
+
+Nested Submenus:
+
+An app can now request to add a submenu to another submenu by specifying a parentID. This did not require any new parameters on the HMI side. HMI's should be updated to process `parentID` in an AddSubMenu request. This param was previously only reserved for AddCommands.
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:92]:
```suggestion
An app can now request to add a submenu to another submenu by specifying a `parentID`. This did not require any new parameters on the HMI side. HMIs should be updated to process `parentID` in an `AddSubMenu` request. This param was previously only reserved for `AddCommands`. 
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+...
+    <param name="menuParams" type="Common.MenuParams" mandatory="true">
+        <description>Position, parentID, and name of menu to be added.</description>
+    </param>
+...
+```
+
+Dynamic Menus:
+
+Two new RPCs were added to HMI API: UI.OnUpdateFile and UI.OnUpdateSubmenu. UI.OnUpdateFile request allows the HMI to request images from an SDL connected app when needed in an effort to reduce the amount of data an app needs to save on the headunit. UI.OnUpdateSubmenu request allows the HMI to dynamically request when submenu information is populated by the app. This functionality helps reduce the system load when an app first connects as the app is not required to load all menu contents onto the head unit immediately.
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:105]: Should code blocks be added around the RPCs here?

Also, "head unit" should be two words:
```suggestion
Two new RPCs were added to HMI API: UI.OnUpdateFile and UI.OnUpdateSubmenu. UI.OnUpdateFile request allows the HMI to request images from an SDL connected app when needed in an effort to reduce the amount of data an app needs to save on the head unit. UI.OnUpdateSubmenu request allows the HMI to dynamically request when submenu information is populated by the app. This functionality helps reduce the system load when an app first connects as the app is not required to load all menu contents onto the head unit immediately.
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    </param>
+    <param name="subMenuDepth" type="Integer" minvalue="1" mandatory="false">
+        <description>The depth of submenus allowed when the driver is distracted. e.g. 3 == top level menu -> submenu -> submenu; 1 == top level menu only</description>
+    </param>
+</struct>
+```
+
+#### New UI Component: Subtle Alert
+
+SubtleAlert RPC was added as a less intrusive UI notification when compared to the Alert RPC. The OnSubtleAlertPressed notification was also added as a way for mobile apps to be aware of and optionally take action when a user clicks on a SubtleAlert notification.
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:169]:
```suggestion
`SubtleAlert` RPC was added as a less intrusive UI notification when compared to the `Alert` RPC. The `OnSubtleAlertPressed` notification was also added as a way for mobile apps to be aware of and optionally take action when a user clicks on a `SubtleAlert` notification.
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+    </description>
+    <param name="appID" type="Integer" mandatory="true">
+        <description>ID of application that is related to this RPC.</description>
+    </param>
+</function>
+```
+
+### Webengine Projection Support
+
+A new `WEB_VIEW` appHMIType and template layout was added which will allow apps to render a template independent view in a browser environment with javascript and HTML.
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:220]:
```suggestion
A new `WEB_VIEW` `appHMIType` and template layout was added which will allow apps to render a template-independent view in a browser environment with JavaScript and HTML.
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+
+```
+<enum name="AppHMIType">
+  <description>Enumeration listing possible app types.</description>
+...
+   <element name="WEB_VIEW" />
+</enum>
+```
+
+This appHMIType must be explicitly specified in an app's policy table entry for SDL Core to allow it to be used.
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:230]:
```suggestion
This `appHMIType` must be explicitly specified in an app's policy table entry for SDL Core to allow it to be used.
```

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+        ],
+        "RequestType": [],
+        "RequestSubType": []
+    },
+...
+```
+
+
+
+### New UI.GetCapabilities parameter: `pcmStreamCapabilities`
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:254]: In previous headers (ie: Deprecated HMI RPC: OnFindApplications) , you didn't include code blocks. For consistency, I think you should either add code blocks to params in other headers, or remove them from this one.

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+
+## Additions
+
+### Vehicle Data
+
+- FuelRange was expanded to replace `fuelLevel` and `fuelLevel_state` parameters
+- New vehicle data type: GearStatus to replace `prndl` parameter
+- New vehicle data type: StabilityControlStatus
+- New vehicle data type: WindowStatus
+- New vehicle data type: HandsOffSteering
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:80]: Should code blocks be added around the new vehicle data?

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
+# Updating from 6.1 to 7.0
```
Comment (theresalech) [inline docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md:1]:
```suggestion
# Migrating SDL Core 6.1 to 7.0
```
I think this should match the file name and config reference.

**Review by theresalech — CHANGES_REQUESTED** [review]
Hi @JackLivio, I've left a few comments and suggested changes. Please let me know if you have any questions!

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md (hunk 10 above)
Comment (JackLivio) [reply docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md]: added

File: docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md (hunk 9 above)
Comment (JackLivio) [reply docs/Migrating to Newer SDL Versions/Migrating SDL Core 6.1 to 7.0/index.md]: fixed

**Pull request merged**